src/System/Hworker.hs

```diff
@@ -302,43 +366,122 @@ worker hw =
+                                             \if del == 1 then\n\
+                                             \  redis.call('lpush', KEYS[2], ARGV[1])\n\
+                                             \end\n\
+                                             \return nil"
+                                             [progressQueue hw, jobQueue hw]
+                                             [t])
+                           delayAndRun
+
+                         Just batch -> do
+                           withNil hw
+                                     (R.eval "local del = redis.call('hdel', KEYS[1], ARGV[1])\n\
+                                             \if del == 1 then\n\
+                                             \  redis.call('lpush', KEYS[2], ARGV[1])\n\
+                                             \  redis.call('hincrby', KEYS[3], 'retries', '1')\n\
+                                             \end\n\
+                                             \return nil"
+                                             [progressQueue hw, jobQueue hw, batchCounter hw batch]
+                                             [t])
+                           delayAndRun
+
                   Failure msg ->
                     do hwlog hw ("Failure: " <> msg)
-                       withNil hw
-                                (R.eval "local del = redis.call('hdel', KEYS[1], ARGV[1])\n\
-                                        \if del == 1 then\n\
-                                        \  redis.call('lpush', KEYS[2], ARGV[1])\n\
-                                        \  redis.call('ltrim', KEYS[2], 0, ARGV[2])\n\
-                                        \end\n\
-                                        \return nil"
-                                        [progressQueue hw, failedQueue hw]
-                                        [t, B8.pack (show (hworkerFailedQueueSize hw - 1))])
-                       void $ R.runRedis (hworkerConnection hw)
-                                         (R.hdel (progressQueue hw) [t])
-                       delayAndRun
+                       case maybeBatch of
+                         Nothing -> do
+                           withNil hw
+                                    (R.eval "local del = redis.call('hdel', KEYS[1], ARGV[1])\n\
+                                            \if del == 1 then\n\
+                                            \  redis.call('lpush', KEYS[2], ARGV[1])\n\
+                                            \  redis.call('ltrim', KEYS[2], 0, ARGV[2])\n\
+                                            \end\n\
+                                            \return nil"
+                                            [progressQueue hw, failedQueue hw]
+                                            [t, B8.pack (show (hworkerFailedQueueSize hw - 1))])
+                           void $ R.runRedis (hworkerConnection hw)
```

Comment:
This part is pretty much unchanged from before. But I'm still trying to figure out why this is here. Isn't it redundant with the lua script immediately above it?

Reply:
Looks like, lol, totally redundant. Git blame got me the commit that introduced the lua script (and the failedQueue) -- the code below already existed and wasn't removed. 

https://github.com/positiondev/hworker/commit/cbeca5862d9a355279a19dfd7461fc620f3ddc33

It's probably been just doing nothing ever since.

Reply:
lol, I see